.circleci/config.yml

```diff
@@ -1269,6 +1269,46 @@ jobs:
           name: Verify Values Match
           command: |
             ./ops/scripts/ci-match-values-between-files.sh "<< parameters.file1_path >>" "<< parameters.pattern_file1 >>" "<< parameters.file2_path >>" "<< parameters.pattern_file2 >>"
+
+  test-proofs:
+    machine: true
+    steps:
+      - checkout
+      - check-changed:
+          patterns: contracts-bedrock/test/kontrol,contracts-bedrock/src/L1/OptimismPortal\.sol
+      - run:
+          name: Checkout submodules
+          command: |
+            git submodule sync
+            git submodule update --init --recursive
+      - run:
+          name: "Start Docker Container"
+          command: |
+            export KONTROL_VERSION=$(cat deps/kontrol_release)
+            docker run --name optimism-ci \
+                    --rm \
+                    -v $(pwd)/kout/proofs:/home/user/workspace/packages/contracts-bedrock/kout/proofs/ \
+                    --interactive \
+                    --tty \
+                    --detach \
+                    --user root \
+                    --workdir /home/user/workspace \
+                    runtimeverificationinc/kontrol:ubuntu-jammy-${KONTROL_VERSION}
+            docker cp . optimism-ci:/home/user/workspace
+            docker exec optimism-ci chown -R user:user /home/user
+      - run:
+          name: 'Run Kontrol'
+          command: |
+            docker exec -u user optimism-ci bash -c 'cd packages/contracts-bedrock && ./test/kontrol/kontrol/run-kontrol.sh'
+      - run:
+          name: Create Results Tar
+          command: |
+            tar -czvf results.tar.gz kout/proofs
+      - store_artifacts:
+          path: results.tar.gz
+          destination: results
+          when: always
```

Comment:
`Property 'when' is not allowed`, based on the schema [here](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/circleciconfig.json)

Reply:
What should it be substituted with? Or should it only be removed? I'm a bit lost here

Reply:
I've removed it. Following [this](https://circleci.com/docs/artifacts/#uploading-core-files), we might want to add `when: on_fail` after ensuring that step works correctly. But adding the `on_fail` flag is not necessary I would say

Reply:
So currently the artifacts are always uploaded, not just on failure? If so I think that's ok so I'll resolve this thread